src/vcpkg/commands.build.cpp

```diff
@@ -1488,54 +1488,75 @@ namespace vcpkg
         return res;
     }
 
-    struct CreateLogDetails
+    static void append_log(const Filesystem& fs, const Path& path, int max_size, std::string& out)
     {
-        const Filesystem& fs;
+        max_size -= static_cast<int>(path.native().size()) + 80 /* size of all static text */;
+        if (max_size < 100)
+        {
+            return;
+        }
+        out += "<details><summary>";
+        out += path.native();
+        out += "</summary>\n\n```\n";
 
-        std::string operator()(const Path& path) const
+        const auto max_log_length = static_cast<size_t>(max_size);
+        const auto start_block_length = max_log_length * 4 / 12;
```

Comment:
What is the significance of these constants?

Reply:
They decide how much we want to see from the start and the end of the log. In general the end is more interesting.

Reply:
I don't mean what they're used for, I can see that part. What I mean is where did 4 and 12 come from?

Reply:
Hm I have tested some values and stayed with this one 😅

Reply:
I see, OK! Just that they have to add up to 12?

Reply:
Yeah. Maybe I change the whole logic and search from start_block_max_length to the begin and from (size - end_block_max_length) to end for a new line and use a fixed cutoff of 1000(?)